IBX-9030: Add content name pattern examples (#2513)
ezsystems/developer-documentation: docs/administration/content_organization/content_types.md

    Note that even if your content type defines a Field intended as a name for the content item (for example, a title of an article or product name), do not confuse it with this Name, which is a piece of metadata, not a Field.

**Identifier** –
an identifier for internal use in configuration files, templates, PHP code, etc.

> It must be unique, can only contain lowercase letters, digits and underscores;
> the maximum length is 50 characters. (Mandatory.)

Author: ```suggestion
It must be unique, can only contain lowercase letters, digits, and underscores.
The maximum length is 50 characters. (Mandatory.)
```

**Description** –
a detailed description of the content type. (Optional.)

<a id="content-name-pattern"></a>
**Content name pattern** –
a pattern that defines what name a new content item based on this content type gets.
The pattern usually consists of Field identifiers that tell the system which Fields it should use when generating the name of a content item.
Each Field identifier has to be surrounded with angle brackets.

> Text outside the angle brackets will be included literally.

Author: ```suggestion
Text outside the angle brackets is included literally.
```

??? note "Pattern examples"

    The following pattern takes the value of the field with the identifier `title` (which is required):
    ```
    <title>
    ```

    The following pattern combines several field values:
    ```
    <firstname> <lastname>
    ```

    The following pattern takes the value of the field with the identifier `seo_title` (which is optional) if not empty,
    else the value of the field with the identifier `short_title` (which is optional),
    else it takes the one with identifier `title` (which is required):
    ```
    <seo_title|short_title|title>
    ```

    The following pattern takes the value of the field with the identifier `nickname` if not empty,
    else it takes the ones with identifiers `firstname` and `lastname` with a space inbetween:
    ```
    <nickname|(<firstname> <lastname>)>
    ```

    - Input-output example:
        - `fistname`: "*Alice*"
        - `lastname`: "*Doe*"
        - `nickname`: "" (empty)
        - Generated content name: "*Alice Doe*"
    - Input-output example:
        - `fistname`: "*Robert*"
        - `lastname`: "*Doe*"
        - `nickname`: "*Bob*"
        - Generated content name: "*Bob*"

>     **Notice that you won't be able to obtain a vertical bar `|`  or parentheses in the generated names as they are special characters.**

Reviewer: ```suggestion
    **The grouping`(`, `)` symbols used together are treated as special keywords and can only be used together as such, without the possibility to use them as standard characters. **
```

Nitpick: believe you can obtain them **in the generated names** - if the fields contain them, they just cannot be part of the pattern in other meaning than the "special one"

I did some tests and it looks like you can use the `|` symbol in its standard meaning - only the grouping is not allowed.

CT settings:
<img width="1333" alt="Zrzut ekranu 2024-10-9 o 13 57 21" src="https://github.com/user-attachments/assets/7168cf63-74a9-49e5-a642-f591a6e74c02">

Content:
<img width="879" alt="Zrzut ekranu 2024-10-9 o 13 57 29" src="https://github.com/user-attachments/assets/ce50af1a-3160-4333-9186-65f459ed5f40">

Author: @mnocon 
OK for removing pipe and being clearer about parentheses.
I'm adding an example of what could go wrong so one could even search for "EZMETAGROUP_0" and find this explanation. What do you think about the following phrasing?:

```suggestion
    **Notice that the grouping parentheses (characters `(`and `)`) are always treated as special characters, and can't be used as standard characters (unexpected text would appear in between like `(EZMETAGROUP_0)`).**
```

Author: Since ibexa/core#449 the whole notice is false (at least on 5.0.7+).

```suggestion
```

    The old URL aliases will continue to redirect to the same content items.

**Container** –
a flag which indicates if content items based on this content type are allowed to have sub-items or not
(mainly relevant for actions via the UI, not validated by every PHP API).

!!! note

    This flag was added for convenience and only affects the interface.
    In other words, it doesn't control any actual low-level logic,
    it simply controls the way the graphical user interface behaves.

**Sort children by default by** –
rule for sorting sub-items.
If the instances of this content type can serve as containers,

> their children will be sorted according to what is selected here.

Author: ```suggestion
their children are sorted according to what is selected here.
```
